spec/Appendix A -- Notation Conventions.md

```diff
@@ -216,3 +216,34 @@ Fibonacci(number):
 Note: Algorithms described in this document are written to be easy to
 understand. Implementers are encouraged to include equivalent but optimized
 implementations.
+
+## Data Collections
+
+This specification describes the semantic properties of data collections using
+types like "list", "set" and "map". These describe observable data collections
+such as the result of applying a grammar and the inputs and outputs of
+algorithms. They also describe unobservable data collections such as temporary
+data internal to an algorithm. Each data collection type defines the operations
+available, and whether values are unique or ordered.
+
+**List**
+
+:: The term _list_ describes a sequence of values which may not be unique. A
+list is ordered unless explicitly stated otherwise (as an "unordered list"). For
```

Comment:
I am not sure "unordered list" makes sense. I mean the ability to iterate over something in well defined way, meaning it can be viewed as a sequence of elements, implies an order in my view.

You could maybe argue that unordered list means the order could change with each iteration, but not sure that is helpful in our context.

Maybe we can get rid of "unordered list"?

Reply:
We don't have any examples of unordered lists being referenced the spec currently, as far as I know. Though [the List wrapper type](https://spec.graphql.org/draft/#sec-List) only dictates that the _serialized_ result is an ordered list, it doesn't specifically, as far as I know, require that the underlying list is ordered.

A common example of an unordered list that might be exposed through GraphQL would be the result of "select * from my_table;" in SQL - this SQL statement has no implicit ordering, and thus may be returned in any order, each time you issue the query you may get a different result even without the underlying data having changed.

I think we should reserve space to reference this concept.